File: 프로그래머스연습문제/문자열을 정수로 바꾸기.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

int solution(string s) {
    int answer = 0;
    int mark = 1;
    int number = 0;
    for(int i = 0; i < s.size(); i++){
        if(s[i] == '-')mark = -1;
        
        else if (s[i] == '+')mark = 1;
        
        else{
            number = s[i] - '0';
            answer +=  number * pow(10,(s.size()-i-1));
        }
    }
    answer = answer * mark;
    return answer;
}

File: 프로그래머스_python/연습문제/Level1/문자열을 정수로 바꾸기.py
def solution(s):
    answer = 0
    
    flag = True
    length = len(s)
    if s[0] == '-':
        flag = False
    else:
        flag = True
        
    for i in s:
        if i.isdigit() == False:
            length -= 1
            continue
        else:
            answer += int(i) * pow(10,length-1)
            length -= 1
    
    if flag == False:
        answer *= -1
    
    return answer
